### PR TITLE
Allow presubmit run multiple sctips

### PR DESCRIPTION
**Pull request by chaodaiG**
Fix: #1746 

/assign @chizhg 
/assign @coryrc

### CODE REVIEW

File: scripts/presubmit-tests.sh [hunk 1]
```diff
@@ -368,7 +368,9 @@ function main() {
       header "Documentation only PR, skipping running custom test"
       exit 0
     fi
-    ${TEST_TO_RUN} || failed=1
+    for test_to_run in ${TESTS_TO_RUN[@]}; do
+      ${test_to_run} || failed=1
```
Comment (coryrc) [inline scripts/presubmit-tests.sh]: can you change "failed=1" to something like { failed=1; echo "--- FAIL: Test ${test_to_run}"; }?
Here and all uses of lone failed=1?

Comment (chaodaiG) [reply scripts/presubmit-tests.sh]: I think it's a good idea. Updated and made a new Spyglass entry so that these messages could be highlighted

**Comment by chaodaiG** [comment]
@coryrc, another lgtm is needed
@chizhg , please take a look at spyglass change

**Comment by coryrc** [comment]
/lgtm

I thought an ending ; was needed in the {} but I guess not

**Comment by chaodaiG** [comment]
> /lgtm
> 
> I thought an ending ; was needed in the {} but I guess not

You're right, it's needed

**Comment by chaodaiG** [comment]
Now it works: https://github.com/knative/serving/pull/7135
@coryrc

File: scripts/presubmit-tests.sh [hunk 2]
```diff
@@ -368,17 +369,19 @@ function main() {
       header "Documentation only PR, skipping running custom test"
       exit 0
     fi
-    ${TEST_TO_RUN} || failed=1
+    for test_to_run in ${TESTS_TO_RUN[@]}; do
```
Comment (chizhg) [inline scripts/presubmit-tests.sh]: Could you update the document for this change?

File: scripts/presubmit-tests.sh [hunk 3]
```diff
@@ -179,7 +179,7 @@ function default_build_test_runner() {
   for pkg in ${tagged_pkgs}; do
     # `go test -c` lets us compile the tests but do not run them.
     if ! capture_output "${report}" go test -c -tags="${tags}" ${pkg} ; then
-      failed=1
+      failed=1;
```
Comment (chizhg) [inline scripts/presubmit-tests.sh]: Not sure what's the best practice for shell, but it seems we do not need this `;`?

File: scripts/presubmit-tests.sh [hunk 4]
```diff
@@ -211,19 +211,19 @@ function run_unit_tests() {
   local failed=0
   # Run pre-unit tests, if any
   if function_exists pre_unit_tests; then
-    pre_unit_tests || failed=1
+    pre_unit_tests || { failed=1; step_failed "pre_unit_tests"; }
```
Comment (chizhg) [inline scripts/presubmit-tests.sh]: It seems we do not need this `;` in the end either?

Comment (chizhg) [reply scripts/presubmit-tests.sh]: Hmm, I saw your comment above, so this ; in the end is needed?

File: scripts/presubmit-tests.sh (hunk 2 above)
Comment (chaodaiG) [reply scripts/presubmit-tests.sh]: added

File: scripts/presubmit-tests.sh (hunk 4 above)
Comment (chaodaiG) [reply scripts/presubmit-tests.sh]: yep it's needed

File: scripts/presubmit-tests.sh (hunk 3 above)
Comment (chaodaiG) [reply scripts/presubmit-tests.sh]: deleted

**Comment by chizhg** [comment]
/lgtm
Thanks @chaodaiG !

**Pull request merged**